rust/integration-tests/src/diff_tests/all_tests.rs

```diff
@@ -0,0 +1,181 @@
+            &test_context,
+            get_expected_generated_txns,
+        )
+        .await;
+    }
+
+    // Helper function to reduce duplicate code for running tests on all processors
+    async fn run_processor_tests<F>(
+        processor_configs: Vec<TestProcessorConfig>,
+        test_context: &TestContext,
+        get_expected_json_path_fn: F,
+    ) where
+        F: Fn(&str, &str) -> String + Send + Sync + 'static + std::marker::Copy,
+    {
+        let processor_map = get_processor_map();
+        for processor_config in processor_configs {
+            let processor_name = processor_config.config.name();
+            let test_type = TestType::Diff(DiffTest);
+
+            if let Some(test_helper) = processor_map.get(processor_name) {
+                let test_helper = Arc::clone(test_helper);
+
+                test_context
+                    .run(
+                        processor_config,
+                        test_type,
+                        move |conn: &mut PgConnection, txn_version: &str| {
+                            let mut json_data = match test_helper.load_data(conn, txn_version) {
+                                Ok(data) => data,
+                                Err(e) => {
+                                    eprintln!(
+                                        "[ERROR] Failed to load data for processor {} and transaction version {}: {}",
+                                        processor_name, txn_version, e
+                                    );
+                                    return Err(e);
+                                }
+                            };
+
+                            let expected_json_path = get_expected_json_path_fn(processor_name, txn_version);
+                            let expected_json = match read_and_parse_json(&expected_json_path) {
+                                Ok(json) => json,
+                                Err(e) => {
+                                    eprintln!(
+                                        "[ERROR] Error handling JSON for processor {} and transaction version {}: {}",
+                                        processor_name, txn_version, e
+                                    );
+                                    return Err(e);
+                                }
+                            };
+
+                            remove_inserted_at(&mut json_data);
+                            // Remove the Err handling around assert_json_eq!, as it doesn't return a Result type
+                            assert_json_eq!(&json_data, &expected_json);
+
+                            println!(
+                                "[INFO] Test passed for processor {} and transaction version: {}",
+                                processor_name, txn_version
+                            );
+                            Ok(())
+                        },
+                    )
+                    .await
+                    .unwrap();
```

Comment:
if one test breaks, it early returns right? do we want to finish all and get a list of results?

Reply:
yeah we can early return! will change

Reply:
yeah that makes sense